cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/DefaultStore.java

```diff
@@ -582,6 +582,17 @@ public RunRecordDetail getRun(ProgramReference programRef, String runId) {
     });
   }
 
+  @Override
+  public void markApplicationsLatest(Collection<ApplicationId> appIds) throws IOException, ConflictException {
+    TransactionRunners.run(transactionRunner, context -> {
+      AppMetadataStore mds = getAppMetadataStore(context);
+      for (ApplicationId appId : appIds) {
+        ApplicationMeta appMeta = getApplicationMeta(mds, appId);
```

Comment:
as mentioned before, there is no need to read the entire row, deserialize it into an ApplicationMeta, then 'create' a new version. We only need to set isLatest to false for the existing latest and set isLatest to true for whatever was specified. There should be a new method added that does this.

Reply:
Added a new method to just update the latest fields in the spec table.